edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/publisher/IoTHubMetricsUpload.cs

```diff
@@ -2,34 +2,44 @@
 
 namespace Microsoft.Azure.Devices.Edge.Agent.Diagnostics.Publisher
 {
+    using System;
     using System.Collections.Generic;
     using System.Linq;
     using System.Threading;
     using System.Threading.Tasks;
     using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Edge.Agent.Core.Requests;
+    using Microsoft.Azure.Devices.Edge.Agent.IoTHub;
     using Microsoft.Azure.Devices.Edge.Util;
 
     public sealed class IoTHubMetricsUpload : IMetricsPublisher
     {
-        readonly ModuleClient moduleClient;
+        readonly IEdgeAgentConnection edgeAgentConnection;
 
-        public IoTHubMetricsUpload(ModuleClient moduleClient)
+        public IoTHubMetricsUpload(IEdgeAgentConnection edgeAgentConnection)
         {
-            this.moduleClient = Preconditions.CheckNotNull(moduleClient, nameof(moduleClient));
+            this.edgeAgentConnection = Preconditions.CheckNotNull(edgeAgentConnection, nameof(edgeAgentConnection));
         }
 
         public async Task PublishAsync(IEnumerable<Metric> metrics, CancellationToken cancellationToken)
         {
             Preconditions.CheckNotNull(metrics, nameof(metrics));
             byte[] data = MetricsSerializer.MetricsToBytes(metrics).ToArray();
-            byte[] compressedData = Compression.CompressToGzip(data);
 
             // TODO: add check for too big of a message
-            if (compressedData.Length > 0)
+            if (data.Length > 0)
```

Comment:
Should these checks go in the module client instead?

Reply:
No, not sending an empty message is specific to this code. Sometimes an empty body is valid.